### PR TITLE
fix(studio): limit customization type filter to supported types (ASTD-488)

### PR DESCRIPTION
**Pull request by walston**
| Before | After |
| --- | --- |
| Screenshot 2026-09-14 at 11 57 32 | Screenshot 2026-09-14 at 11 57 35 |

## Summary

The Custom Models "Finetuning Type" filter (Filter panel on the Models → Fine-tune page) listed all 24 values of the platform `FinetuningType` enum — most of which the product cannot produce (QLoRA, DoRA, PPO, DPO, GRPO, Prefix Tuning, KTO, …). Before: 24 options, ~19 unsupported and unable to match any real custom model. After: only the supported set — LoRA, All Weights, LoRA Merged.

## Changes

- `constants.ts`: source the filter options from the customizer API training-spec enums (`AutomodelTrainingSpecFinetuningType` ∪ `UnslothTrainingSpecFinetuningType` ∪ `RlGRPOTrainingFinetuningType`, deduped) instead of the broad platform `FinetuningType` enum. Labels come from the existing `formatFinetuningType`, so the filter stays in sync with the backend on SDK regen.
- Collapsed the two duplicate option constants (`FINETUNING_TYPE_OPTIONS` + `FINETUNING_TYPE_FILTER_OPTIONS`) into one, dropping the misused `children` key that was only ever remapped to `label`.
- Added `constants.test.ts` locking the option set to the supported types and asserting unsupported approaches are excluded.

## Type of Change

- [x] Code change (feature, bug fix, or refactor)
- [ ] Code change with documentation updates
- [ ] Documentation only
- [ ] Contributor tooling or automation
- [ ] CI, build, or test infrastructure

## Quality Gates

- [x] Tests added or updated for changed behavior
- [ ] Existing tests cover changed behavior — justification:
- [ ] Tests not applicable — justification:
- [ ] Documentation updated for user-visible behavior
- [x] Documentation not applicable — justification: no user docs surface for this filter's option list; behavior is self-evident in the UI.

## Verification

- [x] Pull request title follows the repository's Conventional Commit format
- [x] Every commit includes an appropriate `Signed-off-by:` trailer
- [ ] `uv run pre-commit run -a` passes, or any blocked checks are identified below
- [x] Targeted tests pass, or tests are marked not applicable above
- [x] No secrets, API keys, or credentials are included

Targeted validation:

- `pnpm --filter nemo-studio-ui test src/components/dataViews/CustomModelsDataView/constants.test.ts` → 2 passed.
- `pnpm --filter nemo-studio-ui typecheck` (`tsc --noEmit`) → clean.
- `eslint` on both changed files → clean.
- Commit-time pre-commit hooks (UI lint-staged: eslint + prettier, copyright headers, DCO) passed on the changed files. Full `uv run pre-commit run -a` was not run for this web-only, two-file change.
- Verified live in Studio: the filter dropdown now lists only LoRA / All Weights / LoRA Merged.

## Summary by CodeRabbit

- **Bug Fixes**
  - Updated custom model filters to display only supported fine-tuning types.
  - Removed duplicate fine-tuning options and excluded unsupported values.
  - Improved fine-tuning type labels for consistent display.

- **Tests**
  - Added coverage verifying supported filter options, labels, uniqueness, and exclusion of unsupported values.

### CODE REVIEW

File: web/packages/studio/src/components/dataViews/CustomModelsDataView/constants.test.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025-2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { FINETUNING_TYPE_FILTER_OPTIONS } from '@studio/components/dataViews/CustomModelsDataView/constants';
+
+describe('FINETUNING_TYPE_FILTER_OPTIONS', () => {
+  // ASTD-488: the filter must offer only the finetuning types a completed
+  // customization can actually produce — the union of the customizer backends'
+  // supported values — not the broad platform `FinetuningType` enum.
```
Comment (steramae-nvidia) [inline web/packages/studio/src/components/dataViews/CustomModelsDataView/constants.test.ts]: CAn we remove the comment and issue ref?

**Review by steramae-nvidia — APPROVED** [review]

File: web/packages/studio/src/components/dataViews/CustomModelsDataView/constants.test.ts (hunk 1 above)
Comment (walston) [reply web/packages/studio/src/components/dataViews/CustomModelsDataView/constants.test.ts]: Done — removed the comment and issue ref in 4f759f75bc.
